embassy-stm32/src/dma/dma.rs

```diff
@@ -578,18 +613,194 @@ impl<'a, C: Channel, W: Word> DoubleBuffered<'a, C, W> {
         let ch = self.channel.regs().st(self.channel.num());
         unsafe { ch.ndtr().read() }.ndt()
     }
+}
 
-    pub fn blocking_wait(mut self) {
+impl<'a, C: Channel, W: Word> Drop for DoubleBuffered<'a, C, W> {
+    fn drop(&mut self) {
+        self.request_stop();
         while self.is_running() {}
 
         // "Subsequent reads and writes cannot be moved ahead of preceding reads."
         fence(Ordering::SeqCst);
+    }
+}
 
-        core::mem::forget(self);
+// ==============================
+
+impl<C: Channel> DmaCtrl for C {
+    fn ndtr(&self) -> usize {
+        let ch = self.regs().st(self.num());
+        unsafe { ch.ndtr().read() }.ndt() as usize
+    }
+
+    fn get_complete_count(&self) -> usize {
+        let dma = self.regs();
+        let channel_num = self.num();
+        let index = self.index();
+        // Manually process tcif in case transfer was completed and we are in a higher priority task.
```

Comment:
All the existing drivers already have this "problem", Making the interrupt a lower priority than the task that uses it is not supported. I wouldn't take measures to try to mitigateit.

Reply:
The reason I made this was that I at some point called `get_complete_count()` from a uart interrupt, and the priority relation between the uart and the dma controller is not strict. I can see now that I do not do this anymore, and so this could probably be removed.